src/app/qgsnewspatialitelayerdialog.cpp

```diff
@@ -274,10 +244,22 @@ void QgsNewSpatialiteLayerDialog::selectionChanged()
 
 bool QgsNewSpatialiteLayerDialog::createDb()
 {
-  QString dbPath = mDatabaseComboBox->currentText();
+  QString dbPath = QFileDialog::getSaveFileName( this, tr( "New SpatiaLite Database File" ),
+                   QDir::homePath(),
+                   tr( "SpatiaLite" ) + " (*.sqlite *.db *.sqlite3 *.db3 *.s3db)", nullptr, QFileDialog::DontConfirmOverwrite );
+
   if ( dbPath.isEmpty() )
     return false;
 
+  if ( !dbPath.endsWith( QLatin1String( ".sqlite" ), Qt::CaseInsensitive ) &&
```

Comment:
Could use QgsFileUtils::ensureFileNameHasExtension here (that supports a list of valid extensions)